test/server.cpp

```diff
@@ -322,9 +322,64 @@ TEST_F(ServerTest, 404)
     EXPECT_EQ(404, zfs1_->GET(url)->status) << "url: " << url;
 }
 
+namespace TestingOfHtmlResponses
+{
+
+struct ExpectedResponseData
+{
+  const std::string bookName, bookTitle, expectedBody;
+};
+
+enum ExpectedResponseDataType
+{
+  book_name,
+  book_title,
+  expected_body
+};
+
+// Operator overloading is used as a means of defining a mini-DSL for
+// defining test data in a concise way (see usage in
+// TEST_F(ServerTest, 404WithBodyTesting))
+ExpectedResponseData operator==(ExpectedResponseDataType t, std::string s)
+{
+  switch (t)
+  {
+    case book_name:           return ExpectedResponseData{s, "", ""};
+    case book_title:          return ExpectedResponseData{"", s, ""};
+    case expected_body:       return ExpectedResponseData{"", "", s};
+    default: assert(false); return ExpectedResponseData{};
+  }
+}
+
+std::string selectNonEmpty(const std::string& a, const std::string& b)
```

Comment:
For general usage that might be unsafe. Consider the following example:

```c++
const std::string& x = selectNonEmpty("abc", "");
// x is invalid since it was bound to a temporary object (std::string("abc")) that no longer exists
```

Of course, current usage of `selectNonEmpty()` doesn't contain such bugs. Yet neither does it appear to be in need of optimization at the expense of potential future maintenance headache.